core/sdk-core/src/main/java/software/amazon/awssdk/core/waiters/WaiterOverrideConfiguration.java

```diff
@@ -55,9 +57,25 @@ public Optional<Integer> maxAttempts() {
     }
 
     /**
-     * @return the optional {@link BackoffStrategy} that should be used when polling the resource
+     * @return the optional {@link software.amazon.awssdk.core.retry.backoff.BackoffStrategy} that should be used when polling
+     * the resource
+     * @deprecated Use instead {@link #backoffStrategy2()}
      */
-    public Optional<BackoffStrategy> backoffStrategy() {
+    public Optional<software.amazon.awssdk.core.retry.backoff.BackoffStrategy> backoffStrategy() {
```

Comment:
Can we move the logic to ctor? Should we validate that only of them is specified?

Reply:
It the new backoff strategy is given, what would this one return in that case? (it's just a cosmetic issue, this class while public, is not actually used by customers, rather is just used internally for the waiter executor) You could return  an empty value in this method for that case (not sure if that might confuse someone but for sure will reduce the complexity of it) and that should be fine. If we go that route we won't have any use for the `NonLegacyToLegacyAdapter` and can also go away.

Reply:
We cant move it all to the ctor, as we want to keep a reference to the new BackoffStrategy as a class member